uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeavePeriodEntitlementTest.php

```diff
@@ -365,4 +371,81 @@ public function testGetEntitlementCanReturnTheEntitlementsForLeavePeriodEntitlem
 
     $this->assertEquals($expected, $result);
   }
+
+  public function testGetReturnsOnlyDataLinkedToLoggedInUserWhenUserIsNotALeaveApproverOrAdmin() {
+    $contactID1 = 1;
+    $contactID2 = 2;
+
+    $this->registerCurrentLoggedInContactInSession($contactID1);
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = ['access AJAX API'];
+
+    LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contactID1,
+      'type_id' => 1,
+      'period_id' => 1
+    ]);
+
+    LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contactID2,
+      'type_id' => 1,
+      'period_id' => 1
+    ]);
+
+    $result = civicrm_api3('LeavePeriodEntitlement', 'get', ['check_permissions' => true, 'sequential' => 1]);
+    $this->assertEquals(1, $result['count']);
+    $this->assertEquals($contactID1, $result['values'][0]['contact_id']);
+  }
+
+  public function testGetReturnsOnlyDataLinkedToContactsThatLoggedInUserManagesWhenLoggedInUserIsALeaveApprover() {
+    $manager = ContactFabricator::fabricate();
+    $contact1 = ContactFabricator::fabricate();
+    $contact2 = ContactFabricator::fabricate();
+
+    $this->registerCurrentLoggedInContactInSession($manager['id']);
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = ['access AJAX API'];
+
+    $this->setContactAsLeaveApproverOf($manager, $contact2);
+
+    LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact1['id'],
+      'type_id' => 1,
+      'period_id' => 1
+    ]);
+
+    LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact2['id'],
+      'type_id' => 1,
+      'period_id' => 1
+    ]);
+
+    $result = civicrm_api3('LeavePeriodEntitlement', 'get', ['check_permissions' => true, 'sequential' => 1]);
+    $this->assertEquals(1, $result['count']);
+    $this->assertEquals($contact2['id'], $result['values'][0]['contact_id']);
+  }
+
+  public function testGetReturnsAllDataWhenLoggedInUserHasViewAndEditAllContactsPermission() {
+    $adminID = 1;
+    $contactID1 = 2;
+    $contactID2 = 3;
+
+    $this->registerCurrentLoggedInContactInSession($adminID);
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = ['access AJAX API', 'view all contacts', 'edit all contacts'];
```

Comment:
It would be nice to have separate tests for 'view all contacts' and 'edit all contacts'. What do you think?